mobile/components/PopUpReserva.tsx

```diff
@@ -85,6 +91,42 @@ function PopUpReserva({ onClose, name, fieldId, sport, location, images, descrip
 		return teams.some((team) => team.id === renter.id);
 	}
 
+	const { data: currentReview } = getCurrentUserFieldReview(supabase, fieldId, user?.id!, { enabled: !!user?.id });
+	const currentRating = currentReview?.rating ?? 0;
+
+	useEffect(() => {
+		if (currentReview) {
+			setRating(currentRating);
+		}
+	}, [currentReview]);
+
+	const { data } = getFieldReviewsAvg(supabase, fieldId);
+	type Review = { rating: number };
+	const average = (
```

Comment:
The code in general is really good but asides from this being computed on the server and not the client (let's ignore that 🤏 detail... [it's a security problem waiting to happen] ). The average should be computed as [ current * (n-1) + new ] / n. See: https://math.stackexchange.com/a/106314.

I'll approve these changes anyways since it doesn't make sense for this project, but in the future, scanning ***all*** records of a table to retrieve their value, add them up, and take the average is not the most efficient thing ever.